### ISSUE TITLE
Data source: recipient from integration center

### ISSUE THREAD

**Issue by kvrhdn** [honeycombio/terraform-provider-honeycombio#8]
A data source to refer to existing recipients from the integration center.

Both triggers and SLO's support specifying recipients. Specifying an email recipient is fairly straightforward, but to use Slack you have to use the _recipient ID_.
This data source would be a convenient way to get this ID.

Documentation
- https://docs.honeycomb.io/api/triggers/#specifying-recipients
- https://docs.honeycomb.io/working-with-your-data/triggers/#trigger-recipient-integrations-slack-pagerduty-webhooks

Psuedo Terraform code:

```hcl
data "honeycombio_recipient" "slack" {
  type   = "slack"
  target = "#alerts"
}

data "honeycombio_recipient" "pagerduty" {
  type   = "pagerduty"
}

resource "honeycombio_slo" "example" {
  # ...
  burn_alert {
    # ...
    recipients = [
      honeycombio_recipient.slack.id,
      honeycombio_recipient.pagerduty.id,
    ]
  }
}

resource "honeycombio_trigger" "example" {
  # ...
  recipients = [
    honeycombio_recipient.slack.id,
  ]
}
```

**Comment by kvrhdn** [comment]
Very similar functionality has been added with `honeycombio_trigger_recipient` #16. But `honeycombio_trigger_recipient` is more limited because it has to use the triggers API to query for recipients indirectly. It is for example not possible to add a Slack recipient if there isn't already another trigger with a Slack recipient.
Having API access to the integration center would allows us to add Slack recipients directly.

**Comment by garthk** [comment]
> Having API access to the integration center would allows us to add Slack recipients directly

Webhooks too, please! 👍

**Comment by jharley** [comment]
Closing this as the functionality shipped with [v0.8.0](https://github.com/honeycombio/terraform-provider-honeycombio/releases/tag/v0.8.0) and the new recipient resources and data sources.
